### PR TITLE
Consume IsVersionFromRSM Property

### PR DESCRIPTION
**Pull request by nagworld9**
## Description

Now GS sends IsVersionFromRSM property in agent family to tell requested version is from RSM or not. We use this flag and fall back to old behavior of agent update in non-rsm case.

The schema would be 

**ExtensionConfig.xml**

```
<GAFamily>
 <IsVersionFromRSM>false</IsVersionFromRSM>
<Name>Prod</Name>
<Uris>
<Uri>[https://umsal13dz3z2wk343xwz.blob.core.windows.net/568bb00f-455e-32b8-8deb-0e1bf1636254/568bb00f-455e-32b8-8deb-0e1bf1636254_manifest.xml</Uri>](https://umsal13dz3z2wk343xwz.blob.core.windows.net/568bb00f-455e-32b8-8deb-0e1bf1636254/568bb00f-455e-32b8-8deb-0e1bf1636254_manifest.xml%3C/Uri%3E)
<Uri>[https://umsang5g4cckr0t4xrdm.blob.core.windows.net/568bb00f-455e-32b8-8deb-0e1bf1636254/568bb00f-455e-32b8-8deb-0e1bf1636254_manifest.xml</Uri>](https://umsang5g4cckr0t4xrdm.blob.core.windows.net/568bb00f-455e-32b8-8deb-0e1bf1636254/568bb00f-455e-32b8-8deb-0e1bf1636254_manifest.xml%3C/Uri%3E)
<Uri>[https://umsadkjlshzqkmb23hqp.blob.core.windows.net/568bb00f-455e-32b8-8deb-0e1bf1636254/568bb00f-455e-32b8-8deb-0e1bf1636254_manifest.xml</Uri>](https://umsadkjlshzqkmb23hqp.blob.core.windows.net/568bb00f-455e-32b8-8deb-0e1bf1636254/568bb00f-455e-32b8-8deb-0e1bf1636254_manifest.xml%3C/Uri%3E)
</Uris>
<Version>2.10.0.3</Version>
```
</GAFamily>

**Vmsettings.json**

```
"gaFamilies": [
        {
            "name": "Win7",
            "version": "2.7.41491.1089",
            "isVersionFromRSM": true
            "uris": [
                "[https://umsawqtlsshtn5v2nfgh.blob.core.windows.net/94aa54c8-fa97-60bf-6426-fc691f622cd6/94aa54c8-fa97-60bf-6426-fc691f622cd6_manifest.xml"](https://umsawqtlsshtn5v2nfgh.blob.core.windows.net/94aa54c8-fa97-60bf-6426-fc691f622cd6/94aa54c8-fa97-60bf-6426-fc691f622cd6_manifest.xml%22),
                "[https://umsajbjtqrb3zqjvgb2z.blob.core.windows.net/94aa54c8-fa97-60bf-6426-fc691f622cd6/94aa54c8-fa97-60bf-6426-fc691f622cd6_manifest.xml"](https://umsajbjtqrb3zqjvgb2z.blob.core.windows.net/94aa54c8-fa97-60bf-6426-fc691f622cd6/94aa54c8-fa97-60bf-6426-fc691f622cd6_manifest.xml%22),
                "[https://umsavbvncrpzbnxmxzmr.blob.core.windows.net/94aa54c8-fa97-60bf-6426-fc691f622cd6/94aa54c8-fa97-60bf-6426-fc691f622cd6_manifest.xml"](https://umsavbvncrpzbnxmxzmr.blob.core.windows.net/94aa54c8-fa97-60bf-6426-fc691f622cd6/94aa54c8-fa97-60bf-6426-fc691f622cd6_manifest.xml%22)
            ]
        }
    ],
```

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/protocol/restapi.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ def __init__(self, name, version=None):
         self.name = name
         # This is the Requested version as specified by the Goal State, it defaults to 0.0.0.0 if not specified in GS
         self.requested_version_string = VERSION_0 if version is None else version
+        # Set to False if the version from rsm property is not specified in the GS
+        self.is_version_from_rsm = False
```
Comment (narrieta) [inline azurelinuxagent/common/protocol/restapi.py:77]: having is_version_from_rsm and is_requested_version_specified seems redundant/overcomplicated. can we simplify that?

File: azurelinuxagent/ga/agent_update_handler.py [hunk 2]
```diff
@@ -295,23 +295,23 @@ def run(self, goal_state):
             requested_version = self.__get_requested_version(agent_family)
             agent_manifest = None  # This is to make sure fetch agent manifest once per update
             warn_msg = ""
-            if requested_version is None:
+            if requested_version is not None and agent_family.is_version_from_rsm:
```
Comment (narrieta) [inline azurelinuxagent/ga/agent_update_handler.py:298]: why is this even possible? if RSM is not requesting an update then we would simply get the version from the manifest, right?

File: azurelinuxagent/ga/agent_update_handler.py [hunk 3]
```diff
@@ -295,23 +295,23 @@ def run(self, goal_state):
...
+                GAUpdateReportState.report_expected_version = requested_version
+                # Remove the missing requested version warning once requested version becomes available
+                if "Missing requested version" in GAUpdateReportState.report_error_msg:
+                    GAUpdateReportState.report_error_msg = ""
+            else:
                 # Do not proceed with update if self-update needs to download the manifest again with in an hour
                 if not self.__should_agent_attempt_manifest_download():
                     return
-                if conf.get_enable_ga_versioning():  # log the warning only when ga versioning is enabled
+                if conf.get_enable_ga_versioning() and not agent_family.is_version_from_rsm:  # log the warning only when ga versioning is enabled
```
Comment (narrieta) [inline azurelinuxagent/ga/agent_update_handler.py:309]: but aren't you planning on enabling ga versioning? (and hence this warning would always be there until RSM updates are enabled?)

File: azurelinuxagent/ga/agent_update_handler.py [hunk 4]
```diff
@@ -338,8 +338,8 @@ def run(self, goal_state):
                     raise AgentUpdateError("The Agent received a request to downgrade to version {0}, but downgrading to a version less than "
                                            "the Agent installed on the image ({1}) is not supported. Skipping downgrade.".format(requested_version, daemon_version))
 
-                msg = "Goal state {0} is requesting a new agent version {1}, will update the agent before processing the goal state.".format(
-                    self._gs_id, str(requested_version))
+                msg = "Goal state {0} is requesting a new agent version {1} [VersionFromRSM:{2}], will update the agent before processing the goal state.".format(
```
Comment (narrieta) [inline azurelinuxagent/ga/agent_update_handler.py:341]: change "VersionFromRSM" to something like "Is RSM update"?

File: azurelinuxagent/ga/agent_update_handler.py (hunk 3 above)
Comment (nagworld9) [reply azurelinuxagent/ga/agent_update_handler.py:309]: yet, I'll do it in separate PR. We only log the warning when RSM updates are enabled and still missing requested version

**Comment by nagworld9** [comment]
I'm working on new PR

**Pull request closed without merging**